Sources/Task/ExistentialTask.swift

```diff
@@ -10,29 +10,34 @@
 import Deferred
 import Result
 #endif
-import Dispatch
-
-/// A simple function type expressing the cancellation of some operation.
-public typealias Cancellation = () -> Void
+import Foundation
 
 /// A wrapper over any task.
 ///
 /// Forwards operations to an arbitrary underlying future having the same result
 /// type, optionally combined with some `cancellation`.
-public struct Task<SuccessValue> {
+public final class Task<SuccessValue>: NSObject, NSProgressReporting {
```

Comment:
Maybe I should start leaving around `XPLAT` TODOs around for 3.x. This should stay a `struct` on Linux.